include/time_series/internal/base.hxx

```diff
@@ -279,13 +279,20 @@ template <typename P, typename T>
 void TimeSeriesBase<P, T>::monitor_signal()
 {
     constexpr double SLEEP_DURATION_MS = 100;
+    std::shared_ptr<ConditionVariable<P> > local_condition_ptr = condition_ptr_;
 
-    while (!signal_handler::SignalHandler::has_received_sigint() and
+    while (!local_condition_ptr)
+    {
+        local_condition_ptr = condition_ptr_;
+    }
+
+    while (!signal_handler::SignalHandler::has_received_sigint() &&
            !is_destructor_called_)
     {
         real_time_tools::Timer::sleep_ms(SLEEP_DURATION_MS);
     }
     // notify to release locks that could otherwise prevent the application from
     // terminating
-    condition_ptr_->notify_all();
+    rt_printf("Execution interrupted, notify all waiting components.\n");
```

Comment:
I would suggest removing the print.  I quickly tested the PR with my fake finger demo and get it four times there (I guess once per time series that is involved).  I'll leave the decision to you, though.

Reply:
Removing it seems fair :+1:

Reply:
Can you quickly check if it runs ok on your fake demo finger?
And then we merge.